src/main/java/address/controller/PersonOverviewController.java

```diff
@@ -117,7 +119,10 @@ private void initialize() {
      */
     @FXML
     private void handleDeletePersons() {
-        final List<ReadOnlyViewablePerson> selected = personListView.getSelectionModel().getSelectedItems();
+        final List<ReadOnlyViewablePerson> selected = new ArrayList<>(personListView.getSelectionModel()
+                                                                                    .getSelectedItems());
+        selected.removeIf(p -> !(p instanceof  ReadOnlyViewablePerson));
```

Comment:
You're right, it means the returned list from `getSelectionModel.getSelectedItems` does not follow the javadoc `Collection.contains` interface.... zzzz..

Try `if (selected.isEmpty() || selected.stream().filter(Objects::isNull).findAny().isPresent())`